rust/experimental/server/src/store/memory.rs

```diff
@@ -416,7 +423,15 @@ impl Store for MemoryStore {
                     }
                 }
 
-                self.read_partial_data_with_max_size_limit(candidate_blocks, max_size)
+                self.read_partial_data_with_max_size_limit_and_filter(
+                    candidate_blocks,
+                    max_size,
+                    if last_block_id == -1 {
+                        None
```

Comment:
Why the bitmap will be ignored when the last_block_id = -1 ?

Reply:
I may misunderstood you, let me update it